app/services/league/controllers/leagueController.js

```diff
@@ -1,24 +1,50 @@
 const League = require('../models/leagueModel');
 const User = require('../models/userModel');
+const { Portfolio } = require('../models/portfolioModel');
 
-const addPlayerToLeague = async (user, leagueID) => Promise.all([
-    await League.findByIdAndUpdate(
+const addPlayerToLeague = async (user, leagueID) => {
+    const league = await League.findById(
         leagueID,
-        { $addToSet: { playerList: user } },
-        {},
         (err) => {
             if (err) throw err;
         },
-    ),
-    await User.findOneAndUpdate(
-        { username: user },
-        { $addToSet: { leagues: leagueID } },
-        {},
-        (err) => {
-            if (err) throw err;
-        },
-    ),
-]);
+    );
+    return Promise.all([
+        await League.findByIdAndUpdate(
+            leagueID,
+            { $addToSet: { playerList: user } },
+            {},
+            (err) => {
+                if (err) throw err;
+            },
+        ),
+        await League.findByIdAndUpdate(
+            leagueID,
+            {
+                $addToSet: {
+                    portfolioList: new Portfolio({
+                        owner: user,
+                        league: leagueID,
+                        cash: league.settings.balance,
+                        netWorth: league.settings.balance,
+                    }),
+                },
+            },
+            {},
+            (err) => {
+                if (err) throw err;
+            },
+        ),
```

Comment:
We should combine this into one request if possible

Reply:
Done